qiskit/algorithms/amplitude_amplifiers/amplification_problem.py

```diff
@@ -50,14 +50,25 @@ def __init__(
             objective_qubits: If set, specifies the indices of the qubits that should be measured.
                 If None, all qubits will be measured. The ``is_good_state`` function will be
                 applied on the measurement outcome of these qubits.
-            is_good_state: A function to check whether a string represents a good state.
+            is_good_state: A function to check whether a string represents a good state. By default
+                if the ``oracle`` argument has an ``evaluate_bitstring`` method (currently only
+                provided by the :class:`~qiskit.circuit.library.PhaseOracle` class) this will be
+                used, otherwise this kwarg is required and **must** be specified.
```

Comment:
As a required parameter I guess the typehint now is strange since it still says Optional and as a defaulted parameter has None as the default which is now an invalid value. So by default things are invalid.

Reply:
I can remove `Optional` from the typehint, but that's not quite correct either because if you're using a `PhaseOracle` it will default to using it's `evaluate_bitstring` method. It's a weird edge case that type hints aren't really equipped to express. I'll defer to whatever you want to do

Reply:
I don't have any better ideas - the doctring you added is probably the most realistic solution for now.